src/storage/storage.cc

```diff
@@ -1224,4 +1261,31 @@ bool Storage::ReplDataManager::FileExists(Storage *storage, const std::string &d
   return crc == tmp_crc;
 }
 
+[[nodiscard]] rocksdb::ReadOptions Context::GetReadOptions() const {
+  rocksdb::ReadOptions read_options;
+  if (is_txn_mode) read_options.snapshot = snapshot;
+  return read_options;
+}
+
+[[nodiscard]] rocksdb::ReadOptions Context::DefaultScanOptions() const {
+  rocksdb::ReadOptions read_options = storage->DefaultScanOptions();
+  if (is_txn_mode) read_options.snapshot = snapshot;
+  return read_options;
+}
+
+[[nodiscard]] rocksdb::ReadOptions Context::DefaultMultiGetOptions() const {
+  rocksdb::ReadOptions read_options = storage->DefaultMultiGetOptions();
+  if (is_txn_mode) read_options.snapshot = snapshot;
+  return read_options;
+}
+
+void Context::ResetLatestSnapshot() {
```

Comment:
It should be `RefreshLatestSnapshot`? since it will assign a new one after releasing the old one.